### PR TITLE
fix(explain): add missed fields in explain streaming dist plan

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
- add distinct table in stream agg executor
- add read_pk_prefix_len_hint in table's explaination 
## Checklist For Contributors

- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/frontend/planner_test/tests/testdata/nexmark.yaml [hunk 1]
```diff
@@ -783,8 +783,10 @@
...
+     Table 1 { columns: [$expr1, bid_bidder, count_for_agg_call_4, count_for_agg_call_5, count_for_agg_call_6, count_for_agg_call_7], primary key: [$0 ASC, $1 ASC], value indices: [2, 3, 4, 5], distribution key: [0], read_pk_prefix_len_hint: 0 }
+     Table 2 { columns: [$expr1, bid_auction, count_for_agg_call_8, count_for_agg_call_9, count_for_agg_call_10, count_for_agg_call_11], primary key: [$0 ASC, $1 ASC], value indices: [2, 3, 4, 5], distribution key: [0], read_pk_prefix_len_hint: 0 }
```
Comment (st1page) [inline src/frontend/planner_test/tests/testdata/nexmark.yaml:788]: https://github.com/risingwavelabs/risingwave/issues/8541

File: src/frontend/src/utils/stream_graph_formatter.rs [hunk 2]
```diff
@@ -81,7 +81,7 @@ impl StreamGraphFormatter {
         let tb = TableCatalog::from(tb.clone());
         writeln!(
             f,
-            " Table {} {{ columns: [{}], primary key: {:?}, value indices: {:?}, distribution key: {:?}{} }}",
+            " Table {} {{ columns: [{}], primary key: {:?}, value indices: {:?}, distribution key: {:?}, read_pk_prefix_len_hint: {:?}{} }}",
```
Comment (richardchien) [inline src/frontend/src/utils/stream_graph_formatter.rs:84]:
```suggestion
            " Table {} {{ columns: [{}], primary key: {:?}, value indices: {:?}, distribution key: {:?}, pk prefix len hint: {}{} }}",
```

File: src/frontend/src/utils/stream_graph_formatter.rs [hunk 3]
```diff
@@ -139,23 +140,26 @@ impl StreamGraphFormatter {
...
+                )
+            }),
+            stream_node::NodeBody::Materialize(node) => Some(format!(
+                "materialized table: {}",
+                self.add_table(node.get_table().unwrap())
+            )),
+            stream_node::NodeBody::GlobalSimpleAgg(inner) => {
+                let in_fields = &node.get_input()[0].fields;
+                Some(format!(
+                    "result table: {}, state tables: [{}], distinct_tables: [{}]",
```
Comment (richardchien) [inline src/frontend/src/utils/stream_graph_formatter.rs:159]:
```suggestion
                    "result table: {}, state tables: [{}], distinct dedup tables: [{}]",
```

File: src/frontend/src/utils/stream_graph_formatter.rs [hunk 4]
```diff
@@ -164,12 +168,26 @@ impl StreamGraphFormatter {
...
-                    "result table: {}, state tables: [{}]",
-                    self.add_table(node.get_result_table().unwrap()),
-                    node.agg_call_states
+                        .join(", "),
+                    inner
+                        .get_distinct_dedup_tables()
+                        .iter()
+                        .sorted_by_key(|(i, _)| *i)
+                        .map(|(i, table)| format!(
+                            "(distinct key: {}, table_id: {})",
```
Comment (richardchien) [inline src/frontend/src/utils/stream_graph_formatter.rs:177]:
```suggestion
                            "(distinct key: {}, table id: {})",
```

File: src/frontend/src/utils/stream_graph_formatter.rs [hunk 5]
```diff
@@ -164,12 +168,26 @@ impl StreamGraphFormatter {
...
+                            in_fields[*i as usize].name,
+                            self.add_table(table)
+                        ))
+                        .join(", "),
+                ))
+            }
+            stream_node::NodeBody::HashAgg(inner) => {
+                let in_fields = &node.get_input()[0].fields;
+                Some(format!(
+                    "result table: {}, state tables: [{}], distinct_tables: [{}]",
```
Comment (richardchien) [inline src/frontend/src/utils/stream_graph_formatter.rs:187]:
```suggestion
                    "result table: {}, state tables: [{}], distinct dedup tables: [{}]",
```

File: src/frontend/src/utils/stream_graph_formatter.rs [hunk 6]
```diff
@@ -178,52 +196,63 @@ impl StreamGraphFormatter {
...
-                )),
-                _ => None,
-            };
+                        .join(", "),
+                    inner
+                        .get_distinct_dedup_tables()
+                        .iter()
+                        .sorted_by_key(|(i, _)| *i)
+                        .map(|(i, table)| format!(
+                            "(distinct key: {}, table_id: {})",
```
Comment (richardchien) [inline src/frontend/src/utils/stream_graph_formatter.rs:205]:
```suggestion
                            "(distinct key: {}, table id: {})",
```

**Review by richardchien — APPROVED** [review]
LGTM!

**Review by chenzl25 — APPROVED** [review]

**Pull request merged**